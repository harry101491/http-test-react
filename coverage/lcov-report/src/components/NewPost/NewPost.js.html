<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/components/NewPost/NewPost.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> / <a href="index.html">src/components/NewPost</a> NewPost.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">12.5% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>1/8</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">14.29% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>1/7</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">12.5% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>1/8</span>
      </div>
    </div>
  </div>
  <div class='status-line low'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React, { Component } from 'react';
import axios from 'axios';
&nbsp;
import './NewPost.css';
&nbsp;
class NewPost extends Component {
    state = {
        title: '',
        content: '',
        author: 'Max'
    }
&nbsp;
    postDataHandler = <span class="fstat-no" title="function not covered" >()</span> =&gt; {
        const data = <span class="cstat-no" title="statement not covered" >{</span>
            title: this.state.title,
            body: this.state.content,
            author: this.state.author,
        };
&nbsp;
<span class="cstat-no" title="statement not covered" >        axios</span>
            .post('posts/', data)
            .then(<span class="fstat-no" title="function not covered" >re</span>s =&gt; {
<span class="cstat-no" title="statement not covered" >                console.log('the succesfull response', res);</span>
            })
            .catch(<span class="fstat-no" title="function not covered" >()</span> =&gt; {
<span class="cstat-no" title="statement not covered" >                console.log('response is not coming');</span>
            });
    }
&nbsp;
    render () {
        return (
            &lt;div className="NewPost"&gt;
                &lt;h1&gt;Add a Post&lt;/h1&gt;
                &lt;label&gt;Title&lt;/label&gt;
                &lt;input type="text" value={this.state.title} onChange={<span class="fstat-no" title="function not covered" >(e</span>vent) =&gt; <span class="cstat-no" title="statement not covered" >this.setState({title: event.target.value})}</span> /&gt;
                &lt;label&gt;Content&lt;/label&gt;
                &lt;textarea rows="4" value={this.state.content} onChange={<span class="fstat-no" title="function not covered" >(e</span>vent) =&gt; <span class="cstat-no" title="statement not covered" >this.setState({content: event.target.value})}</span> /&gt;
                &lt;label&gt;Author&lt;/label&gt;
                &lt;select value={this.state.author} onChange={<span class="fstat-no" title="function not covered" >(e</span>vent) =&gt; <span class="cstat-no" title="statement not covered" >this.setState({author: event.target.value})}</span>&gt;
                    &lt;option value="Max"&gt;Max&lt;/option&gt;
                    &lt;option value="Manu"&gt;Manu&lt;/option&gt;
                &lt;/select&gt;
                &lt;button
                    onClick={this.postDataHandler}
                &gt;Add Post&lt;/button&gt;
            &lt;/div&gt;
        );
    }
}
&nbsp;
export default NewPost;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Sun Jun 23 2019 22:36:43 GMT-0700 (Pacific Daylight Time)
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
</body>
</html>
