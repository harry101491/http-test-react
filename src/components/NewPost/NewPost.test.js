import React from 'react';
import { shallow } from 'enzyme';
import NewPost from './NewPost';

describe('<NewPost />', () => {
    let wrapper;
    
    beforeEach(() => {
        wrapper = shallow(<NewPost />);
    });

    it('render as expected', () => {
        expect(wrapper).toMatchSnapshot();
    });
});