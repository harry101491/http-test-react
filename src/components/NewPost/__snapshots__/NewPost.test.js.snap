// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NewPost /> render as expected 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NewPost />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          Add a Post
</h1>,
        <label>
          Title
</label>,
        <input
          onChange={[Function]}
          type="text"
          value=""
/>,
        <label>
          Content
</label>,
        <textarea
          onChange={[Function]}
          rows="4"
          value=""
/>,
        <label>
          Author
</label>,
        <select
          onChange={[Function]}
          value="Max"
>
          <option
                    value="Max"
          >
                    Max
          </option>
          <option
                    value="Manu"
          >
                    Manu
          </option>
</select>,
        <button
          onClick={[Function]}
>
          Add Post
</button>,
      ],
      "className": "NewPost",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Add a Post",
        },
        "ref": null,
        "rendered": "Add a Post",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Title",
        },
        "ref": null,
        "rendered": "Title",
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "onChange": [Function],
          "type": "text",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Content",
        },
        "ref": null,
        "rendered": "Content",
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "onChange": [Function],
          "rows": "4",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "textarea",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Author",
        },
        "ref": null,
        "rendered": "Author",
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <option
              value="Max"
>
              Max
</option>,
            <option
              value="Manu"
>
              Manu
</option>,
          ],
          "onChange": [Function],
          "value": "Max",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Max",
              "value": "Max",
            },
            "ref": null,
            "rendered": "Max",
            "type": "option",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Manu",
              "value": "Manu",
            },
            "ref": null,
            "rendered": "Manu",
            "type": "option",
          },
        ],
        "type": "select",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Add Post",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Add Post",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            Add a Post
</h1>,
          <label>
            Title
</label>,
          <input
            onChange={[Function]}
            type="text"
            value=""
/>,
          <label>
            Content
</label>,
          <textarea
            onChange={[Function]}
            rows="4"
            value=""
/>,
          <label>
            Author
</label>,
          <select
            onChange={[Function]}
            value="Max"
>
            <option
                        value="Max"
            >
                        Max
            </option>
            <option
                        value="Manu"
            >
                        Manu
            </option>
</select>,
          <button
            onClick={[Function]}
>
            Add Post
</button>,
        ],
        "className": "NewPost",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Add a Post",
          },
          "ref": null,
          "rendered": "Add a Post",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Title",
          },
          "ref": null,
          "rendered": "Title",
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "onChange": [Function],
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Content",
          },
          "ref": null,
          "rendered": "Content",
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "onChange": [Function],
            "rows": "4",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "textarea",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Author",
          },
          "ref": null,
          "rendered": "Author",
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <option
                value="Max"
>
                Max
</option>,
              <option
                value="Manu"
>
                Manu
</option>,
            ],
            "onChange": [Function],
            "value": "Max",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Max",
                "value": "Max",
              },
              "ref": null,
              "rendered": "Max",
              "type": "option",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Manu",
                "value": "Manu",
              },
              "ref": null,
              "rendered": "Manu",
              "type": "option",
            },
          ],
          "type": "select",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Add Post",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Add Post",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": false,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
